Runtimes/Core/core/CMakeLists.txt

```diff
@@ -283,9 +283,14 @@ target_compile_options(swiftCore PRIVATE
   "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-Xfrontend -disable-autolinking-runtime-compatibility-concurrency>"
   "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-Xfrontend -disable-objc-attr-requires-foundation-module>"
   "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-Xfrontend -require-explicit-availability=ignore>"
-  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-Xfrontend -prespecialize-generic-metadata>"
-  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-Xllvm -sil-inline-generics>"
-  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-Xllvm -sil-partial-specialization>")
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-Xfrontend -prespecialize-generic-metadata>")
+
+if(NOT "${CMAKE_BUILD_TYPE}" STREQUAL "MinSizeRel")
+  # Using these in MinSizeRel would result in a 15% increase in the binary size
+  target_compile_options(swiftCore PRIVATE
```

Comment:
We could probably keep these in the `target_compile_options` above and let generator expressions handle the condition like so:
```cmake
"$<$<AND:$<COMPILE_LANGUAGE:Swift>,$<NOT:$<CONFIG:MinSizeRel>>>:SHELL:-Xllvm -sil-inline-generics>"
"$<$<AND:$<COMPILE_LANGUAGE:Swift>,$<NOT:$<CONFIG:MinSizeRel>>>:SHELL:-Xllvm -sil-partial-specialization>"
```

But I'm also not going to require that.

Reply:
Thanks for the pointer -- I searched for `CMAKE_BUILD_TYPE` in the documentation of the [generator expressions](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html), and I assumed it was not supported.